setup.py

```diff
@@ -35,24 +36,711 @@
 # stdlib, which is deprecated in Python 3.10 and will be removed in 3.12.
 from distutils import log  # isort: skip
 
+
 # Controls whether providers are installed from packages or directly from sources
 # It is turned on by default in case of development environments such as Breeze
 # And it is particularly useful when you add a new provider and there is no
 # PyPI version to install the provider package from
 INSTALL_PROVIDERS_FROM_SOURCES = 'INSTALL_PROVIDERS_FROM_SOURCES'
-PY39 = sys.version_info >= (3, 9)
+PY39 = sys.version_info >= (
+    3,
+    9,
+)
 
 logger = logging.getLogger(__name__)
 
 version = '2.4.0.dev0'
 
-my_dir = dirname(__file__)
+AIRFLOW_SOURCES_ROOT = Path(__file__).parent.resolve()
+PROVIDERS_ROOT = AIRFLOW_SOURCES_ROOT / "airflow" / "providers"
+
+CROSS_PROVIDERS_DEPS = "cross-providers-deps"
+DEPS = "deps"
+
+# fmt: off
```

Comment:
Possibly we could remove that embedded auto-generated meta-data from setup.py. For now I had some problems in gettting an extra file (airflow/providers_dependencies.json)  tp load while setup was parsed - but have not looked to closely - @uranusjr - is there any limitaion on what we can load here in setup.py ?  I wanted to load AIRFLOW_SOURCES_ROOT / "airflow" / "provider_dependencies.json" here rather than embed it via pre-committ.

Is there any trouble with it ? Should setup.py be mostly standalone or is it ok to loead some files from source tree. Any limitation there (sdist package?, building-in-place?)

Reply:
Ok. Solved it :)